helita/sim/bifrost.py

```diff
@@ -722,8 +1720,23 @@ def write_multi3d(self, outfile, mesh='mesh.dat', desc=None,
         x = self.x[sx] * ul
         y = self.y[sy] * ul
         z = self.z[sz] * (-ul)
-        nh = self.get_hydrogen_pops(sx, sy, sz).to_value('1/cm3')
-        ne = self.get_electron_density(sx, sy, sz).to_value('1/cm3')
+        # if Hion, get nH and ne directly
+        if hion:
```

Comment:
This is old code, please don't add it again. Use the new version with `self.get_hydrogen_pops` and `self.get_electron_density`. (Please update the new `hion` condition inside those functions.)